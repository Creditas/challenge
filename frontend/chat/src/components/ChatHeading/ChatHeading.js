import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const ChatHeading = ({ title }) => (
  <div>
    <AppBar position="static">
      <Toolbar>
        <Typography variant="title" color="inherit">
          { title }
        </Typography>
      </Toolbar>
    </AppBar>
  </div>
);

ChatHeading.propTypes = {
  title: PropTypes.string.isRequired
};

export default ChatHeading;
