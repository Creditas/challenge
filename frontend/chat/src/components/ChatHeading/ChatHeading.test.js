import React from 'react';
import { shallow } from 'enzyme';
import ChatHeading from './ChatHeading';

describe('ChatHeading', () => {
  it('should render', () => {
    const rendered = shallow(<ChatHeading title="Mock" />);
    expect(rendered).toMatchSnapshot()
  })
})
