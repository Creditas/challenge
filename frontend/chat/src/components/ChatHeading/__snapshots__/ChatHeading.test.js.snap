// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeading should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatHeading
    title="Mock"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
>
        <WithStyles(Toolbar)>
                <WithStyles(Typography)
                        color="inherit"
                        variant="title"
                >
                        Mock
                </WithStyles(Typography)>
        </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(Typography)
                    color="inherit"
                    variant="title"
          >
                    Mock
          </WithStyles(Typography)>
</WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            color="inherit"
            variant="title"
>
            Mock
</WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Mock",
            "color": "inherit",
            "variant": "title",
          },
          "ref": null,
          "rendered": "Mock",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
>
          <WithStyles(Toolbar)>
                    <WithStyles(Typography)
                              color="inherit"
                              variant="title"
                    >
                              Mock
                    </WithStyles(Typography)>
          </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(Typography)
                        color="inherit"
                        variant="title"
            >
                        Mock
            </WithStyles(Typography)>
</WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              color="inherit"
              variant="title"
>
              Mock
</WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Mock",
              "color": "inherit",
              "variant": "title",
            },
            "ref": null,
            "rendered": "Mock",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
